lib/AST/Decl.cpp

```diff
@@ -10140,6 +10148,21 @@ MacroRoles swift::getAttachedMacroRoles() {
   return attachedMacroRoles;
 }
 
+bool swift::isMacroSupported(MacroRole role, ASTContext &ctx) {
+  switch (role) {
+  case MacroRole::Expression:
+  case MacroRole::Declaration:
+  case MacroRole::Accessor:
+  case MacroRole::MemberAttribute:
+  case MacroRole::Member:
+  case MacroRole::Peer:
+  case MacroRole::Conformance:
```

Comment:
Oh, I am using it for `codeItem` but I should also use them for freestanding declaration (and all the others even if the feature flags are off), you are right.

Reply:
OK, putting those behind a feature flag failed because then we can’t compile the stdlib anymore. I suspect that the issue is that the features are only enabled if we have `swiftSwiftParser` but we don’t have that yet while building the stdlib during bootstrapping. I just reverted it to always enable non-codeItem macros – freestanding declaration macros don’t have a feature flag (rdar://108637367).